frontend/src/components/input.tsx

```diff
@@ -36,23 +55,43 @@ export const Input = ({ sendMessage, waiting }: InputProps) => {
   return (
     <>
       <form onSubmit={onSend} className={styles.inputContainer}>
-        <input
-          className={styles.input}
-          onChange={onChange}
-          onSubmit={onSend}
+        <textarea
+          className={styles.textarea}
+          ref={textareaRef}
           placeholder="Send a Message..."
-          type="text"
           value={userInput}
+          onChange={onChange}
+          rows={1}
         />
-        <button
-          className={classNames(styles.sendButton, {
+        <div
+          className={classNames(styles.uploadButton_container, {
+            [styles.disabled]: sendDisabled,
+          })}
+        >
+          <button
+            className={classNames(styles.uploadButton, {
+              [styles.disabled]: sendDisabled,
+            })}
+            disabled={sendDisabled}
+          >
+            <img src={UploadIcon} />
+          </button>
+        </div>
+        <div
+          className={classNames(styles.sendButton_container, {
             [styles.disabled]: sendDisabled,
           })}
-          onClick={onSend}
-          disabled={sendDisabled}
         >
-          <img src={RightArrow} />
-        </button>
+          <button
```

Comment:
I would recommend re-using the `button.tsx` component here! I believe in the designs all the buttons of this type behave the same, so hopefully should be able to reuse this component